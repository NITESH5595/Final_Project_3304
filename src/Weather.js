import React, { useState } from "react";
import "./index.css";
import { useNavigate } from "react-router-dom";
import axios from "axios";

function Weather() {
  const [data, setData] = useState({});
  const [location, setLocation] = useState("");

  const url = `https://api.openweathermap.org/data/2.5/weather?q=${location}&units=metric&appid=2f289d6f6e94bd9e23ed6991c58fe88c`;

  const searchLocation = (event) => {
    if (event.key === "Enter") {
      axios.get(url).then((response) => {
        setData(response.data);
        console.log(response.data);
      });
      setLocation("");
    }
  };

  const navigate = useNavigate();
  const goBack = () => {
    navigate("/");
  };

  return (
    <div className="weatherApp">
      <div className="weatherContainer">
        <div className="search">
          <input
            value={location}
            onChange={(event) => setLocation(event.target.value)}
            placeholder="Enter Location"
            type="text "
            onKeyPress={searchLocation}
          />
        </div>
        <div className="top">
          <div className="location">
            <h1>{data.name}</h1>
          </div>
          <div className="temp">
            {data.main ? <h1>{data.main.temp.toFixed()}&deg;C</h1> : null}
          </div>
          <div className="desc">
            {data.weather ? <p>{data.weather[0].main}</p> : null}
          </div>
        </div>
        {data.name != undefined && (
          <div className="bottom">
            <div className="feels">
              {data.main ? (
                <p className="bold">{data.main.feels_like.toFixed()}&deg;C</p>
              ) : null}
              <p>Feels Like</p>
            </div>
            <div className="humidity">
              {data.main ? <p className="bold">{data.main.humidity}%</p> : null}
              <p>Humidity</p>
            </div>
            <div className="wind">
              {data.wind ? (
                <p className="bold">{data.wind.speed.toFixed()}</p>
              ) : null}
              <p>Wind Speed</p>
            </div>
          </div>
        )}
      </div>
    </div>
  );
}

export default Weather;
