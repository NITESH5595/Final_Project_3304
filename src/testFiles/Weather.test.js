import React from "react";
import Weather from "../Weather";
import renderer from "react-test-renderer";

describe("App", () => {
  it("renders without crashing given the required props", () => {
    const wrapper = renderer.create(<Weather />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
