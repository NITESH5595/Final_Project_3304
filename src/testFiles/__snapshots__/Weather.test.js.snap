// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<div
  className="weatherApp"
>
  <div
    className="weatherContainer"
  >
    <div
      className="search"
    >
      <input
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Enter Location"
        type="text "
        value=""
      />
    </div>
    <div
      className="top"
    >
      <div
        className="location"
      >
        <h1 />
      </div>
      <div
        className="temp"
      />
      <div
        className="desc"
      />
    </div>
  </div>
</div>
`;
